src/main/java/com/exadel/sober/controllers/PromiseController.java

```diff
@@ -1,29 +1,55 @@
 package com.exadel.sober.controllers;
 
 import com.exadel.sober.models.Promise;
+import com.exadel.sober.models.Reason;
 import com.exadel.sober.repositories.PromiseRepository;
-import com.exadel.sober.repositories.UserRepository;
-import org.springframework.beans.factory.annotation.Autowired;
+import com.exadel.sober.repositoriesImplemetation.ReasonRepositoryImpl;
+import com.exadel.sober.services.PromiseService;
 import org.springframework.web.bind.annotation.*;
 
+import java.sql.Timestamp;
 import java.util.List;
 
 @RestController
 @CrossOrigin
-@RequestMapping(path="/promise")
+@RequestMapping(path="/promises")
 public class PromiseController {
 
-    @Autowired
     private PromiseRepository promiseRepository;
 
-    @Autowired
-    private UserRepository userRepository;
+    private ReasonRepositoryImpl reasonRepository;
 
-    @PostMapping(path="/deletePromise")
-    public @ResponseBody List<Promise> deletePromiseByPromiseId (@RequestParam Integer promiseId, @RequestParam String userName) {
-
-        promiseRepository.deleteById(promiseId);
-
-        return userRepository.findByName(userName).getPromises();
+    private PromiseService promiseService;
+    public PromiseController(PromiseRepository promiseRepository,
+                             ReasonRepositoryImpl reasonRepository,
+                             PromiseService promiseService) {
+        this.promiseRepository = promiseRepository;
+        this.reasonRepository = reasonRepository;
+        this.promiseService = promiseService;
+    }
+    @GetMapping()
+    public @ResponseBody List<Promise> getPromisesByUserId (@RequestParam Integer userId) {
+        return promiseService.getPromisesForClient(userId);
+    }
+    @GetMapping(path="/getReasons")
+    public @ResponseBody List<Reason> getReasonsByPromiseId (@RequestParam Integer promiseId) {
```

Comment:
I think you should not provide endpoint for reasons because you will have not case to get it separately from Promise. You must get all reasons inside the Promise object when you use getPromisesByUserId()